flocker/node/agents/blockdevice.py

```diff
@@ -617,4 +646,26 @@ def calculate_necessary_state_changes(self, local_state,
             for manifestation
             in manifestations_to_create
         )
+
+        # Issue *another* list_volumes() call.  We could avoid this if we could
+        # smuggle the results from the call made in discover_local_state into
+        # this method call somehow.  That information doesn't fit onto
+        # NodeState, though.  Stuffing it onto self would work but making
+        # discover_local_state have side-effects is probably worse than making
+        # some redundant API calls.  However, in the future, figure out how to
+        # make current_cluster_state carry this information.  It is, after all,
+        # part of the cluster state (and it's wasteful to have every agent list
+        # all the volumes - let alone twice).
+
```

Comment:
Probably `NodeState` needs to grow ability to store backend specific information.